fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/ServerPlayerEvents.java

```diff
@@ -86,6 +110,28 @@ public interface AfterRespawn {
 		void afterRespawn(ServerPlayerEntity oldPlayer, ServerPlayerEntity newPlayer, boolean alive);
 	}
 
+    @FunctionalInterface
+	public interface AllowTeleport {
+		/**
+		 * Called when a player is about to be teleported.
+         * 
+         * @param player the teleporting player
+         * @param pos the new position to teleport to
```

Comment:
I expect the target world may also be useful.

Reply:
Yes, I added it. Thank you!